### ISSUE TITLE
Feature Request - pass down map in context

### ISSUE THREAD

**Issue by ekatzenstein** [uber/react-map-gl#602]
I'm trying to get the mapboxgl map in a child component and it's not available in context (as it is in `react-mapbox-gl`). This is for a custom component that needs to access the layer system. In other words, I'm unable to access the `getMap()` function without specifying a `ref` on the parent component.

**Comment by ekatzenstein** [comment]
This is what I have setup currently:

```
<MapGL
        mapStyle={this.style}
        ref={node => {this.map = node;}}
        {...this.state.viewport}
        mapboxApiAccessToken={MAPBOX_API_PUBLIC_TOKEN}
        onViewportChange={viewport => { this.setState({ viewport });}}
      >
        <CustomComponent map={this.map} />
      </MapGL>
```

For `CustomComponent` to access the mapbox layer `buildings`, this is the only way as far as I can tell, which requires an update after mounting the component.

**Comment by Pessimistress** [comment]
Yes, we can do that. Just curious - what does your custom component do with the buildings layer?

**Comment by ekatzenstein** [comment]
Great, thanks! I'm creating a series of generative geometry components based on `geometry.coordinates` of layers. So not specifically `buildings`, but any layer, referencing the geometry as child of `<MapGL>`

**Comment by Pessimistress** [comment]
Also speaking of past experience dealing with context issues in this module - the map component constructs the `mapboxgl.Map` instance in its `componentDidMount`. Which means `context.map` does not exist yet when `CustomComponent.componentDidMount` is executed. Which means your component cannot use `context.map` as soon as it's available.

You'll still have to manually trigger a rerender for CustomComponent to get the updated context, like:

```
<MapGL ... onLoad={() => this.setState({isMapLoaded: true})}>
  <CustomComponent isMapLoaded={this.state.isMapLoaded} />
<MapGL>
```

**Comment by ekatzenstein** [comment]
Yes, makes complete sense. Would it be worthwhile to pass `isMapLoaded` into context as well for `react-map-gl`? Similar to `isDragging`? I'm trying to minimize required steps for this library

**Comment by ekatzenstein** [comment]
Right, instead of context, passing `isLoaded` to child props would be ideal. Triggers a re-render and confirms when map is loaded. I'm assuming that `isLoaded` would negate the need for checking if `context.map` is null, but that's a trivial step so no problem on the development side.
